fivegmag_5GMSdAwareApplication/app/src/main/java/com/fivegmag/a5gmsdawareapplication/MainActivity.kt

```diff
@@ -187,6 +188,14 @@ class MainActivity : AppCompatActivity(), AdapterView.OnItemSelectedListener {
             requestPermissionLauncher.launch(
                 Manifest.permission.ACCESS_FINE_LOCATION
             )
+        } else if(ActivityCompat.checkSelfPermission(
```

Comment:
This should not be done in an `else if`. In addition, there should be a single request for multiple permissions. Without testing, it should look something like this:

````kotlin
// Step 1: Define the list of permissions to request
val permissions = arrayOf(
    Manifest.permission.ACCESS_FINE_LOCATION,
    Manifest.permission.READ_PHONE_NUMBERS
)

val requestPermissionLauncher =
    registerForActivityResult(
        ActivityResultContracts. RequestMultiplePermissions()
    ) { _: Boolean ->
        initialize()
    }


requestPermissionLauncher.launch(permissions)
````